### PR TITLE
v 0.1.0 - Stable

### PR DESCRIPTION
**Pull request by beardedtim**
## Summary

This PR makes `cms-server` able to be put on NPM, removes some unneeded deps, and cleaned up some errors.

This should be ready for release to NPM as alpha. You can currently test it as follow:

```
$ cd examples

$ cd defaults

$ cp .env.example .env

// fill out the correct MONGO_URI
// usually by changing the `:PORT` inside of `.env` 
// to whatever your Kitematic PORT is.

$ node server.js
```

This should now have a server running at `localhost:5001` or whatever PORT you specified inside of `.env` with a resource available at `/documents`.  The default AUTH_NAME and AUTH_PASS is: `NAME: timi PASS: 1234`

Check out the other examples for more complex ways to configure the application.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,6 +16,8 @@ This is very much a **WIP**. It currently is aiming to do two things:
 
 I am hoping eventually to have this as a working Blogging/page-building platform. As of now, it is easy to add basic endpoints such as `/resource` and `/resource/:id` with little configuration.
 
+*Currently, the server only supports Node v7.9.0.  Checkout [nvm](https://github.com/creationix/nvm) for managing node versions*
```
Comment (beardedtim) [inline README.md]: 👍 

Is there a way we can say "At least 7.9"? Maybe like `7.9^` inside of our `engines` key in `package.json`?

Comment (LanceJernigan) [inline README.md]: Should be taken care of [here](https://github.com/beardedtim/cms-server/pull/4/commits/44873bfba27e720acfe73b4eda27edbf2e585ba0).

File: modules/BaseRoute/index.js [hunk 2]
```diff
@@ -24,41 +23,39 @@ module.exports = (config) => {
...
     ensureAuthorized,
-    cms.addToDB
+    crud.addToDB
   );
 
   Route.delete(
     '/',
     ensureAuthorized,
-    cms.removeCollection(resource)
+    crud.removeCollection(resource)
```
Comment (LanceJernigan) [inline modules/BaseRoute/index.js]: Why do we need to pass the resource here?

Can we not just `drop()` the collection attached to `ctx.collection` like we do in the rest of the crud functions?

Comment (beardedtim) [inline modules/BaseRoute/index.js]: to be honest, the `DELETE /` path is purely for testing as I am not sure I want to offer a way to drop the whole collection via REST in a real application. Should be just remove it from v1?

Comment (LanceJernigan) [inline modules/BaseRoute/index.js]: That's probably not a bad idea.
If someone wants a delete function, it's not too difficult to create one themselves.

Comment (LanceJernigan) [inline modules/BaseRoute/index.js]: Do you want to go ahead and commit those changes?

**Comment by beardedtim** [comment]
Rebased and squashed commits into 1 so if/when merged into master it doesn't show 10+ commits

**Pull request closed without merging**